### PR TITLE
Enhance DJI camera detection and media viewer improvements

### PR DESCRIPTION
**Pull request by DannyJPN**
## Summary

This PR enhances DJI drone camera detection with comprehensive support for all drone models and improves the media viewer functionality.

### DJI Camera Detection Enhancements
- **Encoder tag support**: Extract drone model from video Encoder metadata (DJI NEO, DJIMavic3, DJI M3T)
- **Matrice 300 detection**: Identify Matrice 300 + Zenmuse H20T via filename suffixes:
  - `_T` - Thermal camera
  - `_W` - Wide camera  
  - `_Z` - Zoom camera
  - No suffix - Wide camera
- **DJI Mini 3 detection**: Identify via `Lavf56.15.102` encoder + 4-digit filename pattern
- **Mavic 3 Thermal**: Database lookup for M3T code in INTEGRATED_ENTERPRISE
- **FC3682 fix**: Corrected mapping from Mini 4 Pro to Mini 3
- **Detection priority reordering**: Encoder tag → FC codes → Make tag → Filename patterns

### Media Viewer Improvements
- **Video playback**: OpenCV-based player with standard controls
- **Fire-and-forget mode**: Process media with configurable parallel limits
- **Keyboard shortcuts**: Enhanced UX with intuitive navigation
- **Bunaty WiFi Solar**: Video detection support
- **Realme camera**: Fix duplicate name issue

### Supported DJI Drones
- DJI Mini 3, Mini 3 Pro
- DJI Mavic 3, Mavic 3 Pro, Mavic 3 Thermal
- DJI Matrice 300 + Zenmuse H20T
- DJI Inspire 2 + Zenmuse X5S
- DJI Neo, Avata 2

## Test Plan
- [x] Test DJI Mini 3 detection with `DJI_####.MP4` files
- [x] Test Matrice 300 detection with `_T`, `_W`, `_Z` suffixes
- [x] Test Mavic 3 Thermal detection with M3T encoder
- [x] Test Encoder tag parsing for Neo and Mavic 3
- [x] Test FC code detection for Inspire 2
- [x] Verify no regression in existing camera detection

@codex @claude Please review this PR. I humbly request your peer review and feedback on the implementation.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by DannyJPN** [comment]
## PR Review Issues - Fixed ✅

All critical issues from code review have been addressed:

### ✅ Issue #3: String Indexing Risk (HIGH PRIORITY)
**Fixed:** Added bounds checking before filename suffix extraction
- Added `if len(filename) < 8:` check before `filename[-6:-4]` indexing
- Prevents potential `IndexError` on short filenames
- Location: `exif_camera_detector.py:215-217`

### ✅ Issue #4: FC3682 Test Mismatch (HIGH PRIORITY)
**Fixed:** Corrected test expectation to match actual mapping
- Changed test from expecting "DJI Mini 4 Pro" → "DJI Mini 3"
- Confirmed with user: FC3682 = Mini 3 (no Mini 4 Pro files available)
- Updated test name and docstring for clarity
- Location: `test_dji_camera_mapping.py:29-32`

### ✅ Issue #8: Missing Test Coverage (HIGH PRIORITY)
**Fixed:** Created comprehensive test suite with 31 new tests
- **Encoder tag tests:** DJI NEO, Mavic 3, M3T database lookup
- **Matrice 300 tests:** _T, _W, _Z suffixes + no-suffix detection
- **DJI Mini 3 tests:** Lavf56.15.102 encoder + 4-digit filename
- **FC code tests:** FC3682, FC3582, FC6520
- **Priority tests:** Encoder > FC code > Filename
- **Edge cases:** Short filenames, empty input, bounds checking
- **Other brands:** Samsung, Sony, Realme, Nikon, Bunaty, Canon, Apple

**Test Results:**
- `test_dji_camera_mapping.py`: 21 tests ✅
- `test_exif_camera_detector.py`: 31 tests ✅
- **Total: 52 tests passing**

### 📊 Non-Critical Issues (Acknowledged)
- **Issue #1:** Missing type hint - FALSE POSITIVE (type hints already present)
- **Issue #2:** Error handling - Won't fix (intentional separation for log levels)
- **Issue #5:** Command injection - Low risk (uses list format, not shell=True)
- **Issue #6:** Video memory - Acceptable (cleanup exists in exception handler)
- **Issue #7:** Thread management - Acceptable (daemon threads + alive check)

All critical blockers resolved. PR is now ready for final review and merge.

@codex @claude

**Comment by DannyJPN** [comment]
@claude @codex dokončete peer review s aktuálním stavem.

File: sortunsortedmedia/sortunsortedmediafile.py [hunk 1]
```diff
@@ -15,10 +17,11 @@
 from shared.logging_config import setup_logging
 from shared.exif_handler import get_best_creation_date
 
-from sortunsortedmedialib.constants import DEFAULT_TARGET_FOLDER
+from sortunsortedmedialib.constants import DEFAULT_TARGET_FOLDER, TERMINAL_PAUSE_DURATION, RAW_EXTENSIONS
 from sortunsortedmedialib.media_classifier import classify_media_file
 from sortunsortedmedialib.interactive import ask_for_category
-from sortunsortedmedialib.path_builder import build_target_path, ensure_unique_path
+from sortunsortedmedialib.path_builder import build_target_path, build_edited_target_path, ensure_unique_path
+from sortunsortedmedialib.companion_file_finder import find_jpg_equivalent, find_original_file, extract_metadata_from_path
```
[the remark this answers is not in the document]
Comment (DannyJPN) [reply sortunsortedmedia/sortunsortedmediafile.py]: Missing file in question is present in the master branch,there fore merging is supposed to resolve this issue
